freqtrade/freqtradebot.py

```diff
@@ -195,6 +196,9 @@ def _process(self) -> bool:
             # Refresh whitelist
             self.pairlists.refresh_pairlist()
             self.active_pair_whitelist = self.pairlists.whitelist
+            if not self.active_pair_whitelist:
+                logger.warning('Whitelist is empty.')
+                return False
```

Comment:
This will prevent all open trades from being processed, since `_process` is aborted here.

Removing the return statement wont work either, since that'll duplicate the log-message

Reply:
My point now is that I cannot understand if this implementation is correct or not... If it's not correct -- all the tests are now okay so an additional one should be added that wil fail with current implementation. If it's not correct, then I assume the most straight way is to move both checks into _process() (but actually I guess that DependencyException should be raised earlier, right after fetching whitelist from the config) as you stated above with a pseudocode, but I cannot decide if it should still raise the DependencyException or just issue a log message and return... If you have a decision, could you quickly make changes, even without additional explainations there (I'll dig into details wrt whitelists later), this can be more efficient... This fix is a bit more deep than I can imagine before I created this PR, sorry...

Reply:
Yes, i'll look into it tomorrow if time allows - and yes, i didn't immagine this to go this deep either ...